api/jobs/dto/extended-source-file-dto.ts

```diff
@@ -0,0 +1,7 @@
+import { FullSourceFileDto } from "./full-source-file-dto";
+
+interface ExtendedSourceFileDto extends FullSourceFileDto {
```

Comment:
We already have a mess with `SourceFileDto` vs `FullSourceFileDto` (I didn't check how we use them). Adding one more magic name `ExtendedSourceFileDto` will confuse everyone. I would recommend making it explicit like `JobFileDto`, or something like this.